### ISSUE TITLE
parser: does Not Decode HTML Entities in JSX Attributes (Differs from some other Parsers)

### ISSUE THREAD

**Issue by longlho** [oxc-project/oxc#17807]
## Description

I'm not sure if this is a bug or intentional. 
The OXC parser does not decode HTML entities in JSX attribute values, which differs from the behavior of `@typescript-eslint/parser` and some other JSX/HTML parsing behavior (e.g `acorn`, `babel`, `espree`). This causes inconsistencies when the same code is parsed by different tools.

## Expected Behavior

According to the JSX specification and HTML/XML standards, HTML entities in JSX attribute values should be decoded during parsing:

```jsx
// Input JSX
<FormattedMessage
  defaultMessage="{flightNumber} &middot; {departureCity} ({departureAirport}) to {arrivalCity} ({arrivalAirport})"
/>

// Expected parsed value (what @typescript-eslint/parser returns)
"{flightNumber} · {departureCity} ({departureAirport}) to {arrivalCity} ({arrivalAirport})"  // &middot; decoded to Unicode U+00B7
```

ASTExplorer: https://astexplorer.net/#/gist/750c082fbbc9d121d7f8c63b86240b53/latest
OXC Playground: https://playground.oxc.rs/#eNptVNtq3DAQ/RWjh9LClqaBQtheICRdKOTWpjQveRlbY0eJLLmj8e4G43/vyGtptyT7stKZ25kzIw+qUktVfFl5aoEZ9SWGAA3eu+L/n8Yaesuz+eu9Gmprmge+6tsSaSzetEZrz5+LQWMHxD3hmeHnsXi7B04NdZ54fFewLwYgMmuwyWu+Zp979YLDh29qobxaDop6F/+scayWTD0uVD13QGpZgw2CMIELEc6ICd6CNHmOlQUCNt6FbNw8GMbQQYUZasE1dn+tfNuRCJCBUPlOzDsC4bktvU23qm5mt3GhpP8QiQ0Kt4wuSF1RncNW+gFr/eYXij7uuudgNK56V0VqKVWsibTGGyCJTegU9+fkh2MykrHKhoBCm031ncjTjAqHqNXEQc4HWg2qD/gbyn1TDOWd0fyglscLhU5f1xfGSZfK1kK3k2o820+OpJqJEv3sPe91egzb21fgv/F2Q76TWgrCe4eoUUtOGZQRes2Zb1uINmsFlT5MblZ62aT+xb6B5yAuJUH1hHwrQ5Pw5JxQaHHHfK7vy0es+I6gkxxJ1Fho4nrKImTZC0OkgzBRK+/RTi8GalDWTmE4Pvr4SRKIgudYS4y8ojMLIawMWp0HgtsOybToGKxsnpfFy4MRa2s4o5fIoIFFhHlqr2/soIKQ7WT2SA5k5xLVymtscHobLs44r+Nj0F5e+qymc56nXAmx2CTfKUva87gs6RFI5767wDXmpE+I3ZWonHZnCpV99HYluxkj1kilDwdaGhdnEE35NAosX5dJ8mF/GiMeZTakxn9rmaUw

**Comment by camc314** [comment]
Related: https://github.com/oxc-project/oxc/issues/9667

**Comment by overlookmotel** [comment]
We've known about this bug for some time, but you're seemingly the first to notice it (or at least to open an issue)!

Despite the difference in issue title, this is a repeat of #9667. Closing in favour of that. Thanks for letting us know that this matters!

**Comment by longlho** [comment]
Cool I'll subscribe to the other one. This is especially important in the i18n realm which is why it prob matters to formatjs more.
